app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -215,31 +215,86 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
                 lifecycleOwner = this.viewLifecycleOwner
             )
 
-            toolbarIntegration.set(
-                feature = browserToolbarView.toolbarIntegration,
-                owner = this,
-                view = view
-            )
+            var owner = this
+            browserToolbarView.loadAsync {
```

Comment:
nit: as I mention in my large comment below, I think we should do something like:
```kotlin
browserToolbarView.loadAsync { toolbarView ->
```

And replace references to `browserToolbarView.view` with `toolbarView` for safer binding.